Source/CarthageKit/FrameworkExtensions.swift

```diff
@@ -262,6 +262,38 @@ extension NSURL {
 		return .Failure(.ReadFailed(self, error))
 	}
 
+	public var carthage_absoluteString: String {
+		#if swift(>=2.3)
+			return absoluteString!
+		#else
+			return absoluteString
+		#endif
+	}
+
+	public func appendingPathExtension(pathExtension: String) -> NSURL {
+		#if swift(>=2.3)
+			return URLByAppendingPathExtension(pathExtension)!
+		#else
+			return URLByAppendingPathExtension(pathExtension)
+		#endif
+	}
+
+	public func appendingPathComponent(pathComponent: String) -> NSURL {
+		#if swift(>=2.3)
+			return URLByAppendingPathComponent(pathComponent)!
+		#else
+			return URLByAppendingPathComponent(pathComponent)
+		#endif
+	}
+
+	public func appendingPathComponent(pathComponent: String, isDirectory: Bool) -> NSURL {
```

Comment:
These `appendingPath...` functions are used to work around optionality difference between Swift 2.2 and 2.3, and also match the Swift 3 APIs:

- https://github.com/apple/swift/blob/swift-3.0-PREVIEW-5/stdlib/public/SDK/Foundation/URL.swift#L784
- https://github.com/apple/swift/blob/swift-3.0-PREVIEW-5/stdlib/public/SDK/Foundation/URL.swift#L809
- https://github.com/apple/swift/blob/swift-3.0-PREVIEW-5/stdlib/public/SDK/Foundation/URL.swift#L855